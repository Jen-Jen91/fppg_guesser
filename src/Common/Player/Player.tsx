import React from "react";
import { roundOffPoints } from "../helpers";
import "./Player.scss";

export interface PlayerProps {
  playerFirstName: string;
  playerLastName: string;
  url: string;
  fppg: number;
  showFppg: boolean;
  clickPlayer: (fppg: number) => void;
}

const Player = (props: PlayerProps) => {
  const {
    playerFirstName,
    playerLastName,
    url,
    fppg,
    showFppg,
    clickPlayer
  } = props;

  return (
    <div onClick={() => clickPlayer(fppg)} className="player-container">
      <img
        src={url}
        alt={playerFirstName + " " + playerLastName}
        width={300}
        height={300}
      />

      <p className="name-text">
        {playerFirstName} {playerLastName}
      </p>

      {showFppg && <p className="fppg-text">Fppg: {roundOffPoints(fppg)}</p>}
    </div>
  );
};

export default Player;
