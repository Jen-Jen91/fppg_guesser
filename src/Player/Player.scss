@import "../Common/variables.scss";

.player-container {
  @include flex-column(center, flex-start);

  border: solid 5px $light-blue;
  border-radius: 10px;

  cursor: pointer;

  &:hover {
    background-color: $light-blue;
  }
}

.name {
  margin: 15px 0;

  font-size: 1.5rem;
  font-weight: bold;
  color: $dark-blue;
}

.fppg-text {
  @extend .name;

  margin: 0 0 10px 0;

  font-size: 0.9rem;
  text-transform: uppercase;
}
