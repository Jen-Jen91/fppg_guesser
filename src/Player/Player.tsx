import React from "react";

export interface PlayerProps {
  playerFirstName: string;
  playerLastName: string;
  url: string;
  fppg: number;
  clickPlayer: (fppg: number) => void;
}

export default function Player(props: PlayerProps) {
  const { playerFirstName, playerLastName, url, fppg, clickPlayer } = props;

  return (
    <div onClick={() => clickPlayer(fppg)} className="player-container">
      <img src={url} alt={playerFirstName + " " + playerLastName} />
      <p>
        {playerFirstName} {playerLastName}
      </p>
    </div>
  );
}
