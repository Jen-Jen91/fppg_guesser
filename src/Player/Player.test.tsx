import React from "react";
import Player from "./Player";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

test("Player snapshot test", () => {
  const tree = renderer
    .create(
      <Player
        playerFirstName="John"
        playerLastName="Doe"
        url=""
        fppg={48}
        clickPlayer={() => console.log("Click")}
      />
    )
    .toJSON();
  console.log("TREE: ", tree);
  expect(tree).toMatchSnapshot();
});

test("Player calls clickPlayer when clicked", () => {
  const playerClicked = jest.fn();
  const wrapper = mount(
    <Player
      playerFirstName="John"
      playerLastName="Doe"
      url=""
      fppg={48}
      clickPlayer={playerClicked}
    />
  );

  const div = wrapper.find(".player-container");
  div.simulate("click");
  expect(playerClicked).toBeCalledWith(48);
});
