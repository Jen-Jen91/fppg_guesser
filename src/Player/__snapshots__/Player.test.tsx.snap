// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player snapshot test 1`] = `
<div
  className="player-container"
  onClick={[Function]}
>
  <img
    alt="John Doe"
    src=""
  />
  <p>
    John
     
    Doe
  </p>
</div>
`;
